components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -3030,7 +3029,30 @@ export function createWalletApi () {
                   id: [arg.chainId, arg.contractAddress].join('-')
                 }
               ]
-      })
+      }),
+      generateReceiveAddress: mutation<string, BraveWallet.AccountId>({
```

Comment:
Why is this a mutation and not a query?

Reply:
For ETH/SOL/FIL it is always account's address. For bitcoin it checks if 'next receive address' is still not transacted and tries next one if it is(prefs are changed).
So in general case it is not guaranteed it always returns same address.